src/torcontrol.cpp

```diff
@@ -522,9 +522,9 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
     if (reply.code == 250) {
         LogPrint(BCLog::TOR, "tor: Authentication successful\n");
 
-        // Now that we know Tor is running setup the proxy for onion addresses
-        // if -onion isn't set to something else.
-        if (gArgs.GetArg("-onion", "") == "") {
+        // Now that we know Tor is running, setup the proxy for onion addresses
+        // if -onion and -proxy isn't set to something else.
+        if (gArgs.GetArg("-onion", "-proxy") == "") {
```

Comment:
The default must be set to a value, not the name of the arg it should default to.

Reply:
I think I might have fixed it now.

Reply:
@MarcoFalke I noticed in another spots this was handled differently. Would it be more preferable like this:

```
        // Now that we know Tor is running setup the proxy for onion addresses
        // if -onion isn't set to something else.
        std::string onion = gArgs.GetArg("-onion", "");
        std::string proxy = gArgs.GetArg("-proxy", "");
        if (onion.empty() && proxy.empty()) {
```